src/main/scala/Color.scala

```diff
@@ -22,36 +26,12 @@ sealed trait Color:
   inline def queen  = this - Queen
   inline def king   = this - King
 
-  val white = this == Color.White
-  val black = this == Color.Black
-
-object Color:
-
-  case object White extends Color:
-
-    inline def unary_! = Black
-
-    val passablePawnRank   = Rank.Fifth
-    val promotablePawnRank = Rank.Eighth
-    val backRank           = Rank.First
+  lazy val white = this == Color.White
```

Comment:
We have to use lazy val here because otherwise `white` is initialised before `White`. So, `white` is always false.

More detail here: https://scastie.scala-lang.org/lenguyenthanh/C5WRuTkXQjeF3Xslas4Ayw/49